test/unit/QtiItemAssetCloudFrontReplacerTest.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\awsTools\test\unit;
+
+use Aws\S3\S3Client;
+use GuzzleHttp\Psr7\Stream;
+use League\Flysystem\AwsS3v3\AwsS3Adapter;
+use oat\awsTools\AwsClient;
+use oat\awsTools\items\ItemCloudFrontReplacement;
+use oat\awsTools\QtiItems\QtiItemAssetCloudFrontReplacer;
+use oat\oatbox\filesystem\Directory;
+use oat\oatbox\filesystem\File;
+use oat\oatbox\log\LoggerService;
+use oat\tao\model\media\MediaAsset;
+use oat\tao\model\media\MediaBrowser;
+use oat\taoItems\model\media\ItemMediaResolver;
+use oat\taoItems\model\render\ItemAssetsReplacement;
+use oat\taoQtiItem\model\compile\QtiAssetCompiler\QtiItemAssetCompiler;
+use oat\taoQtiItem\model\compile\QtiAssetReplacer\QtiItemAssetReplacer;
+use oat\taoQtiItem\model\compile\QtiAssetReplacer\QtiItemNonReplacer;
+use oat\taoQtiItem\model\compile\QtiItemCompilerAssetBlacklist;
+use oat\taoQtiItem\model\pack\QtiAssetPacker\PackedAsset;
+use oat\taoQtiItem\model\qti\Img;
+use oat\taoQtiItem\model\qti\Item;
+use oat\taoQtiItem\model\qti\XInclude;
+use oat\taoQtiItem\test\unit\model\compile\mock\ElementMock;
+use oat\generis\test\TestCase;
+use Psr\Log\NullLogger;
+
+class QtiItemAssetCloudFrontReplacerTest extends TestCase
+{
+    /** @var QtiItemAssetCompiler */
```

Comment:
I think this unit test should test `QtiItemAssetCloudFrontReplacer` and the public methods `shouldBeReplaced` and `replace`, not `QtiItemAssetCompiler`